packages/itwin/ec3-widget/src/common/UseEC3WidgetLocalization.ts

```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useMemo } from "react";
+import { EC3WidgetUI } from "../EC3WidgetUI";
+
+export interface useEC3WidgetLocalizationResult {
```

Comment:
This is way too error prone. Look around in this repository to see how other packages implemented localization.

Example: https://github.com/iTwin/viewer-components-react/blob/master/packages/itwin/tree-widget/src/TreeWidget.ts
Usage: https://github.com/iTwin/viewer-components-react/blob/master/packages/itwin/tree-widget/src/components/SelectableTree.tsx#L151

Reply:
changed the approach as discussed on call